R/pkg/R/DataFrame.R

```diff
@@ -1053,6 +1057,13 @@ setMethod("collect",
                   df[[colIndex]] <- col
                 } else {
                   colType <- dtypes[[colIndex]][[2]]
+                  if (is.null(PRIMITIVE_TYPES[[colType]])) {
+                    specialtype <- specialtypeshandle(colType)
+                    if (!is.null(specialtype)) {
+                      colType <- specialtype
```

Comment:
@felixcheung You are right. Here is a mistake. The goal of this function is to change the `Key` for the dictionary PRIMITIVE_TYPES. The issue is because backend returns `decimal(10,0)` as the Key. Let me review the whole PR again and make changes as you suggested. Thanks!